### DISCUSSION TITLE
Display Drivers and DisplayType

### DISCUSSION THREAD

**Discussion by kevinsmgov** [OrchardCMS/OrchardCore#10152] [Q&A]
Is it possible for a Display Driver to change the display type it returns based on a runtime condition?

I tried changing `context.Shape.Metadata.DisplayType` when the ViewModel is populated but it doesn't appear to have any effect.

My use case is a content part that has child content items with it (they have their own display driver). There is a limit property in the parent content part (a NumericField). If the limit is 1 I want the parent and child to both have `DisplayType` set to "Detail". If the limit is greater than 1 I want the  parent and children to both have `DisplayType` set to "Summary". Setting the child items is easy because the parent display driver can set the display type in the `BuildDisplayAsync` call for each child. However, I can't tell how to have the parent driver set its own `DisplayType`.

Any advice welcome.

**Comment by ns8482e** [comment]
I guess you need  to add an alternate - no need to change `DisplayType`.
Inside driver, just add  your conditional alternate using `Displaying`  event
```cs
public override IDisplayResult Display(MyPart myPart, BuildPartDisplayContext context)
        {
         
            return Initialize<MyPartViewModel>("MyPartViewModel", model =>
            {
                // init viewmodel properties
            })
            .Displaying( ctx=> {
               if( myPart.ContentItem.Content.Limit  > 1)
                    ctx.Metadata.Alternates.Add("MyPartViewModel_Multiple");
               else
                    ctx.Metadata.Alternates.Add("MyPartViewModel_Single");
            });
        }
```

and have two templates  `MyPartViewModel.Single.cshtml` and `MyPartViewModel.Multiple.cshtml`

**Reply by kevinsmgov** [reply]
That's a good suggestion. Another factor that I didn't mention is that I'm trying to driver all of the template variation via placement.json (I was hoping I could set DisplayType so placement.json could filter by that and set appropriate results). I ended up using a single template for the parent of the collection and setting a Multiple property in the ViewModel.
